staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -137,15 +139,34 @@ func ObjectReaction(tracker ObjectTracker) ReactionFunc {
 			if err != nil {
 				return true, nil, err
 			}
-			// Only supports strategic merge patch
-			// TODO: Add support for other Patch types
-			mergedByte, err := strategicpatch.StrategicMergePatch(old, action.GetPatch(), obj)
-			if err != nil {
-				return true, nil, err
-			}
-
-			if err = json.Unmarshal(mergedByte, obj); err != nil {
-				return true, nil, err
+			// Only supports strategic merge patch and JSONPatch as coded.
+			// Before adding JSONPatch support, PatchType was ignored and everything
+			// was treated as strategic merge patch. JSONPatch flat out failed,
+			// so support for it has been added, not sure about the merge patch
+			// behaviour, so leaving it as is.
+
+			// obj gets set to the new unmarshalled patched object
+			switch action.GetPatchType() {
+			case types.JSONPatchType:
+				patch, err := jsonpatch.DecodePatch(action.GetPatch())
+				if err != nil {
+					return true, nil, err
+				}
+				modified, err := patch.Apply(old)
+				if err != nil {
+					return true, nil, err
+				}
+				if err = json.Unmarshal(modified, obj); err != nil {
+					return true, nil, err
+				}
+			default:
```

Comment:
In general the patch strategy is specified by the value of the `patchStrategy` key in a field tag.(in a struct) Additionally `kubectl` defaults to `strategic`, I'm not sure if there is server side defaulting ?

I think that the idea here is to return an error if the strategy was not specified because it would be better if the tests were more explicit. Hopefully returning an error by default will not fail other tests :)

Reply:
The real server returns error is the patch type isn't specified: https://github.com/kubernetes/kubernetes/blob/48086eb7d41e6a76830da5c658352e9042ae00bd/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go#L72-L74

> Do you mean, that if the patch type is not strategicmergepatch that I return error and see if tests break?

Yes. If tests break, and it requires significant changes to fix, please let us know. We can fix it in another PR.